kani-compiler/src/codegen_cprover_gotoc/codegen/statement.rs

```diff
@@ -76,17 +76,38 @@ impl<'tcx> GotocCtx<'tcx> {
                 self.codegen_set_discriminant(dest_ty, dest_expr, *variant_index, location)
             }
             StatementKind::StorageLive(var_id) => {
-                if self.queries.args().ignore_storage_markers {
+                if !self.current_fn().is_address_taken_local(*var_id) {
                     Stmt::skip(location)
                 } else {
-                    Stmt::decl(self.codegen_local(*var_id, location), None, location)
+                    let global_dead_object = cbmc::global_dead_object(&self.symbol_table);
```

Comment:
Is the high-level translation approach documented somewhere ?

Maybe just add a comment and example explaining how Mir declarations are mapped to DECL/DEAD and how storage markers are mapped to dynamic updates of dead_object ? Also does it solve any of the problems related to multiple occurrences of StorageLive/StorageDead or the problems with StorageDead not being dominated by StorageLive ?

Reply:
I have added a comment explaining the approach and also stating that this does address the risk of multiple/mismatching occurrences of StorageLive/StorageDead.